test/server/compiler-test.js

```
@@ -409,4 +409,97 @@ describe('Compiler', function () {
                 });
         });
     });
+
+    describe('Raw data compiler', function () {
+        it('Should compile test files', function () {
+            var sources = ['test/server/data/test-suites/raw/test.testcafe'];
+
+            return compile(sources)
+                .then(function (compiled) {
+                    var testfile = resolve('test/server/data/test-suites/raw/test.testcafe');
+                    var tests    = compiled.tests;
+                    var fixtures = compiled.fixtures;
+
+                    expect(tests.length).eql(3);
+                    expect(fixtures.length).eql(2);
+
+                    expect(fixtures[0].name).eql('Fixture1');
+                    expect(fixtures[0].path).eql(testfile);
+                    expect(fixtures[0].page).eql('about:blank');
+
+                    expect(fixtures[1].name).eql('Fixture2');
+                    expect(fixtures[1].path).eql(testfile);
+                    expect(fixtures[1].page).eql('http://example.org');
+
+                    expect(tests[0].name).eql('Fixture1Test1');
+                    expect(tests[0].fixture).eql(fixtures[0]);
+
+                    expect(tests[1].name).eql('Fixture1Test2');
+                    expect(tests[1].fixture).eql(fixtures[0]);
+
+                    expect(tests[2].name).eql('Fixture2Test1');
+                    expect(tests[2].fixture).eql(fixtures[1]);
+                });
+        });
+
+        it('Should raise error if can not parse raw file', function () {
```

Should raise an error if it cannot parse a raw file.